src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] entryArray;
+    private int size;
+
+    public ArrayList() {
+        entryArray = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    private void extendArrayLength() {
+        int oldArrayLength = entryArray.length;
+        int newArrayLength = oldArrayLength + (oldArrayLength >> 1);
+        entryArray = Arrays.copyOf(entryArray, newArrayLength);
+    }
+
     @Override
     public void add(T value) {
-
+        if (entryArray.length <= size) {
+            extendArrayLength();
+        }
+        entryArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("index is too big");
+        }
+        if (entryArray.length <= size) {
+            extendArrayLength();
+        }
+        System.arraycopy(entryArray, index, entryArray, index + 1, size - index);
+        entryArray[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("index is too big");
+        }
```

Comment:
code duplication, extract to a private utility method, include index and size in the message for debugging purposes